### ISSUE TITLE
Crashes on river and amy

### ISSUE THREAD

**Issue by kwalcock** [clulab/fatdynet#5]
The xor example program is crashing on river and amy.  It works OK on clara, Travis, and my computer with Ubuntu 18.04, 16.04, and 14.04.  It appears to be a memory corruption kind of problem that usually appears in devices.cc at DeviceMempoolSizes::DeviceMempoolSizes(const std::string & descriptor).  If I add std::cout statements there, it can get through the function but crashes somewhere else shortly thereafter.  The value of the descriptor is "512" and isn't itself a problem.  To diagnose, I probably need to build on river or amy, run the C++ program, hope to see the same behavior, and catch it with the debugger.

**Comment by MihaiSurdeanu** [comment]
Thanks for catching this!
This is weird, since river runs 16.04 LTS... 
In any case, I think we should release this on maven as is, while we figure this out.

**Comment by kwalcock** [comment]
The problem is that CMake is configured in CMakeLists.txt, line 85 with
```
set(CMAKE_CXX_FLAGS_RELEASE "-funroll-loops -Ofast -march=native")
```
When this is compiled on my relatively new processor, a certain VPSRLQ instruction (and many others, but this is the first problem) is generated.  This is not appreciated on Amy's relatively old Q3'13 Xeon processor or River's AMD processor.  If the architecture is changed to a more generic x86-64, then the program runs on both of the problem machines.  It may run slightly slower elsewhere, but the compatibility is probably worth it.
The change will eventually be documented on the wiki page about how to get the fatjar out of dynet.  I'll be prepared to make a new release, perhaps after receiving reports on whether any other changes are necessary.

**Comment by kwalcock** [comment]
This should be fixed as of #9.
